src/structures/VoiceState.js

```diff
@@ -196,6 +209,76 @@ class VoiceState extends Base {
     return true;
   }
 
+  /**
+   * Request to speak in the channel. Only applicable for stage channels.
+   */
+  async requestToSpeak() {
+    const channel = this.channel;
+    if (!channel || channel.type !== 'stage') throw new Error('VOICE_NOT_STAGE_CHANNEL');
+
+    const selfInChannel = this.member && this.member.voice && this.member.voice.channelID === this.channelID;
+    if (!selfInChannel) throw new Error('VOICE_NOT_IN_CHANNEL');
+
+    if (this.client.user.id !== this.id) throw new Error('VOICE_STATE_NOT_OWN');
+
+    const member = this.member;
+    const hasRequestToSpeakPermission = member && member.permissions.has(Permissions.FLAGS.REQUEST_TO_SPEAK);
+
+    if (!hasRequestToSpeakPermission) throw new Error('VOICE_NEED_REQUEST_TO_SPEAK');
+    await this.client.api
+      .guilds(this.guild.id)('voice-states')('@me')
+      .patch({
+        data: {
+          channel_id: this.channelID,
+          request_to_speak_timestamp: new Date().toISOString(),
+        },
+      });
+  }
+
+  /**
+   * Suppress/unsuppress the user.
+   * @param {boolean} suppressed - Whether or not the user should be suppressed.
+   */
+  async setSuppressed(suppressed) {
+    if (typeof suppressed !== 'boolean') throw new TypeError('VOICE_STATE_INVALID_TYPE', 'suppressed');
+
+    const channel = this.channel;
+    if (!channel || channel.type !== 'stage') throw new Error('VOICE_NOT_STAGE_CHANNEL');
+
+    const member = this.member;
+    const isBot = member && member.user.bot;
+    const isMe = this.client.user.id === this.id;
+
+    const selfInChannel = this.member && this.member.voice && this.member.voice.channelID === this.channelID;
+    if (!selfInChannel) throw new Error('VOICE_NOT_IN_CHANNEL');
+
+    const hasMuteMembersPermission = member && member.permissions.has(Permissions.FLAGS.MUTE_MEMBERS);
```

Comment:
this should check permissions in the channel, not the server

```suggestion
    const hasMuteMembersPermission = member && member.permissionsIn(channel).has(Permissions.FLAGS.MUTE_MEMBERS);
```